### PR TITLE
track Swift version used to compile Realm Swift

### PR DESCRIPTION
**Pull request by jpsim**
This will give us a better idea of when it's reasonable to drop older Swift versions. /cc @tgoyne @bdash

### CODE REVIEW

**Comment by tgoyne** [comment]
cocoapods-setup doesn't call set-swift-version because we'd gone with the explicit separate branches for swift versions with CP, so the file never gets created.

**Comment by jpsim** [comment]
> cocoapods-setup doesn't call set-swift-version because we'd gone with the explicit separate branches for swift versions with CP, so the file never gets created.

Right. I suppose we could write the file based on the user's value of `xcrun swift --version` at `cocoapods-setup` time, although that will only get updated at each `pod install`. If https://github.com/apple/swift-evolution/pull/69 lands in Swift 2.2, then this won't be too problematic.

Can you think of a better way to do this? I wouldn't go so far as installing a build phase to update the file in `cocoapods-setup`.

**Comment by jpsim** [comment]
@tgoyne @bdash CocoaPods fix in place.

File: build.sh [hunk 1]
```diff
@@ -852,6 +858,8 @@ case "$COMMAND" in
             cp -R include/Realm include/realm
           fi
           touch include/Realm/RLMPlatform.h
+        else
+          echo "let swiftLanguageVersion = \"$(get_swift_version)\"" > RealmSwift/SwiftVersion.swift
```
Comment (bdash) [inline build.sh]: Why is this in the `else` block? I'm not sure how this logic relates to the `without-core` argument.

Comment (jpsim) [inline build.sh]: This definitely needs a comment, but it's because `without-core` is only used when installing Realm Swift.

Comment (jpsim) [inline build.sh]: Actually, it makes more sense at this point to rename the `without-core` argument to `swift`.

**Comment by bdash** [comment]
:+1:

File: build.sh [hunk 2]
```diff
@@ -412,6 +412,12 @@ case "$COMMAND" in
         ln -s "RealmSwift-swift$version" RealmSwift
         git update-index --assume-unchanged RealmSwift || true
 
+        SWIFT_VERSION_FILE="RealmSwift/SwiftVersion.swift"
+        CONTENTS="let swiftLanguageVersion = \"$version\""
+        if ! grep "$CONTENTS" "$SWIFT_VERSION_FILE"; then
```
Comment (tgoyne) [inline build.sh]: `grep -q` to avoid noise in the build logs

Comment (jpsim) [inline build.sh]: Good call, thanks.

**Pull request merged**